### DISCUSSION TITLE
Gas and Transactions in Ethereum

### DISCUSSION THREAD

**Discussion by tcrin** [Cyfrin/foundry-full-course-cu#2958] [Q&A] [answered]
Hi everyone,
I have a few questions and would appreciate any clarifications from everyone:

1. A `Transaction` command will be recorded on the Blockchain, so it will be added to a block. Miners are the ones who create the block. Therefore, I will pay the gas fee to the Miner, is that correct? If so, how does the EVM benefit from this transaction, since the EVM also consumes resources during the execution?

2. A `call` command is not recorded on the Blockchain, so it does not incur gas fees. If this is the case, can we still spam using call commands, potentially leading to network congestion? How does the EVM prevent this?

**Comment by EngrPips** [comment, marked as answer]
Hello @tcrin, I will start with the simplest question, the second one.

**ANSWER TO QUESTION 2**
So when you send a call `command` it is just like you are trying to get data from a database, and that is it. The process doesn't involve the network processing any transaction whatsoever as you are specifically getting data from the blockchain node you are connected to, and other blockchain nodes does not even know that you requested that data. So, I don't see how you can actually cause network congestion by requesting data from a single node.

**ANSWER TO QUESTION 1**
I am not sure how to answer your first question because `EVM` simply means `ethereum virtual machine`, which is the machine on which Ethereum smart contract runs on, So I am not very sure what you mean by `how does the EVM benefit from this transaction since the EVM also consumes resources during the execution?` but maybe reading the documentation on the official page of Ethereum will help get more in-depth understanding about `EVM` and Ethereum as a whole.

**Reply by tcrin** [reply]
Thanks a lot @EngrPips,

Regarding question 1, I will read more Ethereum documentation. 

As for question 2, I’ve understood it now, hehe, turns out a call only reads data from the node we are connected to.

**Reply by EngrPips** [reply]
Anytime, Friend. I forgot to share the link to the docs. Here it is https://ethereum.org/en/developers/docs/
